File: exercises/repeatedpattern.rb
def repeatedPattern(str)
    substr = ""
    i = 0
    length = 0
    while(i < (str.length / 2))
        substr.concat(str[i])
        length += 1
        j = i + 1
        while(j < str.length)
            temp = str.slice(j, j + length)
            if temp != substr
                break
            end
            j += length
        end
        i += 1
        if j == str.length 
            return true
        end
    end
    false
end

File: exercises/repeatedpattern.js
function repeatedPattern(str) {
    let substring = "";
    let i = 0;
    let l = 0;
    while(i < str.length / 2) {
      substring += str[i]
      l++
      let j = i + 1
      while(j < str.length) {
        let temp = str.slice(j, j + l)
        if(temp !== substring) {
          break
        }
        j += l
      }
      i++
      if(j === str.length) return true
    }
    return false
  }